app/views/shared/_items_index.html.haml

```diff
@@ -1,22 +1,25 @@
-- @categories.each do |category|
+%article.search
+  - if @q
+    = search_form_for @q do |f|
+      = f.label 'Looking for something?'
+      = f.search_field :name_or_description_or_category_title_cont
+      = f.submit
+
+- if active_search?
   %article.category-header
-    %h1
-      = category.title
+    %h2
+      = Results for \"#{@q.conditions.first.values.first.value}\"
   %article.items-container
-    - category.items.each do |item|
-      - if item.active?
-        %section.item-card{:id => item.name.tr(' ', '-')}
-          %article.item-card-image
-            %a{'href' => item_path(item)}
-              %img{'src' => item.image_url, 'alt' => item.name}
-          %article.item-info
-            %aside.item-title
-              %h3
-                = link_to item.name, item_path(item)
-            %aside.item-price
-              = format_price_per_unit(item)
-          %article.item-description
-            %p
-              = item.description
-          %article.item-card-button
-            = button_to "Add to Cart", carts_path(item_id: item.id)
\ No newline at end of file
+    - if @items.any?
```

Comment:
This view logic looks appropriate to me. Nice use of the `shared/items` partial.